apps/algolia/lib/algolia/update.ex

```diff
@@ -13,14 +13,43 @@ defmodule Algolia.Update do
   """
   @spec update(String.t() | nil) :: t
   def update(host \\ nil) do
-    Map.new(@indexes, &{&1, update_index(&1, host)})
+    Map.new(@indexes, &{&1, update_and_clean_index(&1, host)})
   end
 
-  @spec update_index(atom, String.t()) :: success | error
-  def update_index(index_module, base_url) do
-    index_module.all()
-    |> Enum.filter(&has_routes?/1)
-    |> Enum.map(&build_data_object/1)
+  @spec update_and_clean_index(atom, String.t()) :: :ok | error
+  def update_and_clean_index(index_module, base_url) do
+    new_objects =
+      index_module.all()
+      |> Enum.filter(&has_routes?/1)
+      |> Enum.map(&to_data_object/1)
+
+    with {:ok, _body} <- clean_index(new_objects, index_module, base_url),
+         {:ok, _body} <- update_index(new_objects, index_module, base_url) do
+      :ok
+    end
+  end
+
+  @spec clean_index(Enum.t(), atom, String.t()) :: success | error
+  def clean_index(new_objects, index_module, base_url) do
+    current_objs = current_objects(base_url, index_module)
+
+    current_objs
+    |> filter_objects(new_objects)
+    |> Enum.map(&build_action_object(&1, "deleteObject"))
+    |> build_request_object()
+    |> send_update(base_url, index_module)
+  end
+
+  defp filter_objects(current_objects, new_objects) do
+    Enum.filter(current_objects, fn co ->
```

Comment:
I don't think we should ever use non-descriptive names...`co` for `current_object`.

Reply:
I will change them.  The names were a singular simplification of the enumberables name.  I have always used them when it comes to simple one line loops, as the the enumerable in question is very close to the logic that consumes it.  I agree that non-descriptive names shouldn't be used on more complex loops.